### PR TITLE
[Fabric] Shim RCTPixelValue to take scale factor

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [x] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Fabric on macOS implementation:
Shim RCTPixelValue to take scale factor

## Changelog

[macOS][Added] - [Fabric] Shim RCTPixelValue to take scale factor

## Test Plan

[x] Build RNTester-macOS w/ Fabric - doesn’t run yet, but no RCTPixelValue errors
CleanShot 2022-12-12 at 15 50 53

Build errors: 
[Build RNTester-macOS_2022-12-09T16-25-02.txt](https://github.com/microsoft/react-native-macos/files/10213033/Build.RNTester-macOS_2022-12-09T16-25-02.txt)

[x] Build RNTester - iOS w/ Fabric

[x] Build RNTester-macOS w/ Paper - should work
CleanShot 2022-12-12 at 15 45 36

[x] Build RNTester - iOS w/ Paper - should work
CleanShot 2022-12-12 at 15 44 44

### CODE REVIEW

**Comment by shwanton** [comment]
53587dce8e636d4c7c3d55d02dc88befa2cc6255 will be removed once it's merge into `main`

File: ReactCommon/react/renderer/textlayoutmanager/platform/ios/RCTTextLayoutManager.mm [hunk 1]
```diff
@@ -57,7 +57,12 @@ - (TextMeasurement)measureNSAttributedString:(NSAttributedString *)attributedStr
 
   CGSize size = [layoutManager usedRectForTextContainer:textContainer].size;
 
+#if !TARGET_OS_OSX // [TODO(macOS GH#774)
   size = (CGSize){RCTCeilPixelValue(size.width), RCTCeilPixelValue(size.height)};
+#else
+  CGFloat scale = [[NSScreen mainScreen] backingScaleFactor];;
```
Comment (Saadnajmi) [inline ReactCommon/react/renderer/textlayoutmanager/platform/ios/RCTTextLayoutManager.mm:63]: Double semi colon?

Comment (shwanton) [reply ReactCommon/react/renderer/textlayoutmanager/platform/ios/RCTTextLayoutManager.mm:63]: oof, good catch

**Comment by shwanton** [comment]
closes #1555 #1561

File: React/Fabric/Mounting/ComponentViews/SafeAreaView/RCTSafeAreaViewComponentView.mm [hunk 2]
```diff
@@ -38,12 +38,14 @@ - (UIEdgeInsets)_safeAreaInsets
   return UIEdgeInsetsZero;
 }
 
+#if !TARGET_OS_OSX // [TODO(macOS GH#774)
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/ComponentViews/SafeAreaView/RCTSafeAreaViewComponentView.mm:41]: Comment about blocks here too

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**